src/genie_python/genie.py

```diff
@@ -42,8 +44,9 @@
 )
 from genie_python.version import VERSION  # noqa E402
 
+E = TypeVar("E", bound=np.generic, covariant=True)
 PVBaseValue = bool | int | float | str
-PVValue = PVBaseValue | list[PVBaseValue] | npt.NDArray | None
+PVValue = PVBaseValue | list[PVBaseValue] | npt.NDArray[E] | None
```

Comment:
Hmm I'm not sure about making `PVValue` itself generic... Sorry I might have misunderstood your question a bit earlier. In this case the more pragmatic thing might be to silence the error on this line (which I presume was telling you to add more type-args to `npt.NDArray`?)